backends/bmv2/psa_switch/psaSwitch.cpp

```diff
@@ -1042,8 +1043,61 @@ void ExternConverter_Register::convertExternInstance(
 
 void ExternConverter_Random::convertExternInstance(
     UNUSED ConversionContext* ctxt, UNUSED const IR::Declaration* c,
-    UNUSED const IR::ExternBlock* eb, UNUSED const bool& emitExterns)
-{ /* TODO */ }
+    UNUSED const IR::ExternBlock* eb, UNUSED const bool& emitExterns) {
+
+    if (eb->getConstructorParameters()->size() != 2) {
+      modelError("%1%: expected two parameters", eb);
+      return;
+    }
+
+    auto inst = c->to<IR::Declaration_Instance>();
+    cstring name = inst->controlPlaneName();
+
+    // adding random instance into extern_instances
+    auto jext_mtr = new Util::JsonObject();
+    jext_mtr->emplace("name", name);
+    jext_mtr->emplace("id", nextId("extern_instances"));
+    jext_mtr->emplace("type", eb->getName());
+    jext_mtr->emplace_non_null("source_info", eb->sourceInfoJsonObj());
+    ctxt->json->externs->append(jext_mtr);
+
+    // adding attributes to random extern_instance
+    Util::JsonArray *arr = ctxt->json->insert_array_field(jext_mtr, "attribute_values");
+
+    // range min
+    auto min = eb->findParameterValue("min");
+    CHECK_NULL(min);
+    if (!min->is<IR::Constant>()) {
+        modelError("%1%: expected a constant", min->getNode());
+        return;
+    }
+    auto attr0 = eb->getConstructorParameters()->getParameter(0);
+    auto min_c = min->to<IR::Constant>();
+    auto bitwidth0 = ctxt->typeMap->minWidthBits(min_c->type, min->getNode());
+    cstring val0 = BMV2::stringRepr(min_c->value, ROUNDUP(bitwidth0, 8));
+    auto rmin = new Util::JsonObject();
+    rmin->emplace("name", attr0->toString());
+    rmin->emplace("type", "hexstr");
+    rmin->emplace("value", val0);
+    arr->append(rmin);
+
+    // range max
+    auto max = eb->findParameterValue("max");
+    CHECK_NULL(max);
+    if (!max->is<IR::Constant>()) {
+        modelError("%1%: expected a constant", max->getNode());
+        return;
+    }
+    auto attr1 = eb->getConstructorParameters()->getParameter(1);
+    auto max_c = max->to<IR::Constant>();
+    auto bitwidth1 = ctxt->typeMap->minWidthBits(max_c->type, max->getNode());
+    cstring val1 = BMV2::stringRepr(max_c->value, ROUNDUP(bitwidth1, 8));
+    auto rmax = new Util::JsonObject();
+    rmax->emplace("name", attr1->toString());
+    rmax->emplace("type", "hexstr");
+    rmax->emplace("value", val1);
+    arr->append(rmax);
+}
```

Comment:
Since the min and max values are compile-time known values, we could give a compile-time error if `(min_c->value > max_c->value)`, couldn't we?  That seems better to me than a warning message during execution in psa_switch.

Reply:
Done, thanks :)